Tests/MongoSwiftTests/SpecTestRunner/SpecTest.swift

```diff
@@ -65,9 +65,8 @@ extension SpecTest {
             expect(seenError).to(beNil(), description: self.description)
         }
 
-        if let expectedResult = self.outcome.result {
-            expect(result).toNot(beNil(), description: self.description)
-            expect(result).to(equal(expectedResult), description: self.description)
+        if let expectedResult = self.outcome.result, let receivedResult = result {
```

Comment:
[One of the tests](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/tests/bulkWrite.yml#L287) consistently sees a connection error instead of a bulk write error (which may or may not be intended, I couldn't really tell), so I updated this to only compare results when it did receive a bulk write error. Let me know if you think this is the right approach.

Reply:
Woops, that link was wrong. [This](https://github.com/mongodb/specifications/blob/master/source/retryable-writes/tests/bulkWrite.yml#L212) is the correct one. The failpoint is configured to fail twice which seems to me like this write is supposed to fail, but I'm not sure if it's supposed to fail with a connection error. I'll check with other drivers.

Reply:
So after talking with Jeremy, I have made the following changes:
- `ServerError.bulkWriteError` now has an `otherError` field corresponding to any error that falls outside of the write error / write concern error definitions (e.g. connection)
- The only `ServerError` or `RuntimeError` case that `MongoCollection.bulkWrite` will throw is `.bulkWriteError`, even in the case of connection / auth errors.
    - This is a little undesirable, but it is pretty much the only way to remain consistent when dealing with bulk write errors.
    - We _could_ throw top level other errors when the result is empty and there aren't any write / write concern errors, but, again, this would be pretty inconsistent with how theyd be thrown after anything succeeded.
- We can now just compare results as we originally had it and error if a result isn't present when the spec wants one.
- While looking through this code, I improved the description getter.

Reply:
that all sounds good to me 👍